src/ai/backend/manager/models/keypair.py

```diff
@@ -624,7 +624,18 @@ async def mutate(
         info: graphene.ResolveInfo,
         access_key: AccessKey,
     ) -> DeleteKeyPair:
+        from .user import UserRow
+
         ctx: GraphQueryContext = info.context
+        async with ctx.db.begin_readonly_session() as db_session:
+            user_query = sa.select(UserRow.uuid).where(UserRow.main_access_key == access_key)
+            uuid_list: list[str] = [
+                str(raw_uuid) for raw_uuid in (await db_session.scalars(user_query)).all()
+            ]
+            if uuid_list:
+                return DeleteKeyPair(
+                    False, f"the keypair is used as main access key by user(uuid: {uuid_list})"
+                )
```

Comment:
Seems like these lines can be simplified by just evaluating result of `COUNT(*)` query, as the purpose of this block is to check if the user is utilizing this keypair as its main keypair.